pkg/apis/v1beta1/ec2nodeclass.go

```diff
@@ -113,6 +113,16 @@ type EC2NodeClassSpec struct {
 	// +kubebuilder:default={"httpEndpoint":"enabled","httpProtocolIPv6":"disabled","httpPutResponseHopLimit":2,"httpTokens":"required"}
 	// +optional
 	MetadataOptions *MetadataOptions `json:"metadataOptions,omitempty"`
+	// Tenancy of the instance. An instance with a tenancy of dedicated runs on single-tenant hardware.
+	//
+	// If not set, Tenancy will be set to default.
+	// Currently, Karpenter only support default and dedicated option.
+	// For more information,
+	// See the AWS::EC2::LaunchTemplate Placement Document.
+	// https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-launchtemplate-placement.html
+	// +kubebuilder:validation:Enum:={default,dedicated}
+	// +optional
+	Tenancy *string `json:"tenancy,omitempty"`
```

Comment:
Do we even need default or can we just assume that not specifying this option assumes that you are using "shared"?

Reply:
If not specified, tenancy is "default (shared)".
If wish, We can also specify tenancy: "default" in yaml.